scripts/build_backend.py

```diff
@@ -131,16 +131,32 @@ def custom_build_cmake_clib(
         cmake_args += ["-DADD_ONEDAL_RPATH=ON"]
 
     cpu_count = multiprocessing.cpu_count()
+    # convert to max supported pointer to a memory size in kB
+    memfree = sys.maxsize >> 10
     # limit parallel cmake jobs if memory size is insufficient
-    # TODO: add on all platforms
     if IS_LIN:
         with open("/proc/meminfo", "r") as meminfo_file_obj:
-            memfree = meminfo_file_obj.read().split("\n")[1].split(" ")
-            while "" in memfree:
-                memfree.remove("")
-            memfree = int(memfree[1])  # total memory in kB
-        cpu_count = min(cpu_count, floor(max(1, memfree / 2**20)))
-
+            next(meminfo_file_obj)  # skip MemTotal
+            memfree = int(
+                next(meminfo_file_obj).strip().split()[-2]
+            )  # total free physical memory in kB
+    elif IS_WIN:
+        txt = subprocess.run(
+            [
+                "powershell",
+                "Get-CIMInstance",
+                "Win32_OperatingSystem",
+                "|",
+                "Select",
+                "FreePhysicalMemory",
+            ],
+            stdout=subprocess.PIPE,
+            text=True,
+        )
+        memfree = int(txt.stdout.strip().split()[-1])  # total free physical memory in kB
+
+    mem_per_proc = 20  # 2**20 kB or 1GB
+    cpu_count = min(cpu_count, floor(max(1, memfree >> mem_per_proc)))

```

Comment:
`psutil` will significantly simplify code but introduces new building dependency:
```suggestion
    mem_per_proc = 20  # 2**20 kB or 1GB
    cpu_count = min(psutil.cpu_count(), floor(max(1, psutil.virtual_memory().available >> mem_per_proc)))
```

Reply:
Note that `psutil.cpu_count` can return `None`.